From e02538913885d20574788b95011ae0fef6cee5f7 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Thu, 4 Jan 2024 15:32:42 +0100
Subject: [PATCH 506/841] feat(st-bsec): update OTPCR masks for STM32MP21

BSEC OTPCR register structure is different on STM32MP21.
Adapt register description.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 19e646b370b6fa559b5b2fcf382653d690eab872 feat(st-bsec): update OTPCR masks for STM32MP21]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: Ie837b6e870e9c0980ae91c8e537482e8a0214ebb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/356326
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 include/drivers/st/bsec3_reg.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/drivers/st/bsec3_reg.h b/include/drivers/st/bsec3_reg.h
index 9156b1f36..b2fb7ca44 100644
--- a/include/drivers/st/bsec3_reg.h
+++ b/include/drivers/st/bsec3_reg.h
@@ -36,11 +36,19 @@
 #define BSEC_SIDR			U(0xFFC)
 
 /* BSEC_OTPCR register fields */
+#if STM32MP21
+#define BSEC_OTPCR_ADDR_MASK		GENMASK_32(12, 0)
+#else /* STM32MP21 */
 #define BSEC_OTPCR_ADDR_MASK		GENMASK_32(8, 0)
+#endif /* STM32MP21 */
 #define BSEC_OTPCR_ADDR_SHIFT		0
 #define BSEC_OTPCR_PROG			BIT(13)
 #define BSEC_OTPCR_PPLOCK		BIT(14)
+#if STM32MP21
+#define BSEC_OTPCR_LASTCID_MASK		GENMASK_32(22, 19)
+#else /* STM32MP21 */
 #define BSEC_OTPCR_LASTCID_MASK		GENMASK_32(21, 19)
+#endif /* STM32MP21 */
 #define BSEC_OTPCR_LASTCID_SHIFT	19
 
 /* BSEC_LOCKR register fields */
-- 
2.47.2

